data/StateMachine.j2119

```diff
@@ -166,10 +166,18 @@ A Wait State MAY have a referencePath field named "TimestampPath".
 A Wait State MUST have only one of "Seconds", "SecondsPath", "Timestamp", and "TimestampPath".
 A Wait State MUST have a field named one of "Seconds", "SecondsPath", "Timestamp", or "TimestampPath".
 A Parallel State MUST have an object-array field named "Branches"; each element is a "Branch".
-A Map State MUST have an object field named "Iterator"; its value is a "Branch".
+A Map State MAY have an object field named "Iterator"; its value is a "Branch".
+A Map State MAY have an object field named "ItemProcessor"; its value is a "Processor".
+A Map State MAY have only one of "Iterator" and "ItemProcessor".
 A Map State MAY have a field named "ItemsPath".
+A Map State MAY have an object field named "ItemSelector".
+A Map State MAY have only one of "ItemSelector" and "Parameters".
```

Comment:
The docs say that ItemProcessor and ItemSelector are the replacements for Iterator and Parameters, so I assumed a map should have one or the other, but never both.

Reply:
it should have only one or the other, but it for `ItemProcessor` it MUST have one of ItemProcessor or Iterator